graylog2-web-interface/src/components/event-definitions/event-definition-types/FilterForm.tsx

```diff
@@ -0,0 +1,489 @@
+import type User from 'logic/users/User';
+import useUserDateTime from 'hooks/useUserDateTime';
+import type { EventDefinition } from 'components/event-definitions/event-definition-types';
+import type { Stream } from 'stores/streams/StreamsStore';
+import type { QueryValidationState } from 'views/components/searchbar/queryvalidation/types';
+import { indicesInWarmTier, isSearchingWarmTier } from 'views/components/searchbar/queryvalidation/warmTierValidation';
+
+import EditQueryParameterModal from '../event-definition-form/EditQueryParameterModal';
+import commonStyles from '../common/commonStyles.css';
+
+export const PLUGGABLE_CONTROLS_HIDDEN_KEY = 'pluggableSearchBarControlsAreHidden';
+export const TIME_UNITS = ['HOURS', 'MINUTES', 'SECONDS'];
+
+const LOOKUP_PERMISSIONS = [
+  'lookuptables:read',
+];
+
+const buildNewParameter = (name: string) => ({
+  name: name,
+  embryonic: true,
+  type: 'lut-parameter-v1',
+  data_type: 'any',
+  title: 'new title',
+  // has no binding, no need to set binding property
+});
+
+type Props = {
+  currentUser: User,
+  eventDefinition: EventDefinition,
+  onChange: (name: string, config: EventDefinition['config']) => void,
+  streams: Array<Stream>,
+  validation: { errors: { [key: string]: Array<string> } }
+}
+
+const FilterForm = ({
+  currentUser,
+  eventDefinition,
+  onChange,
+  streams,
+  validation,
+} : Props) => {
+  const { execute_every_ms: executeEveryMs, search_within_ms: searchWithinMs } = eventDefinition.config;
+  const searchWithin = extractDurationAndUnit(searchWithinMs, TIME_UNITS);
+  const executeEvery = extractDurationAndUnit(executeEveryMs, TIME_UNITS);
+  const { userTimezone } = useUserDateTime();
+  const { setFieldWarning, warnings } = useContext(FormWarningsContext);
+  const validationState = (warnings?.queryString) as QueryValidationState;
+  const warmTierRanges = indicesInWarmTier(validationState);
+
+  const { tables } = useStore(LookupTablesStore);
+  const { pathname } = useLocation();
+
+  const sendTelemetry = useSendTelemetry();
+
+  const queryId = generateId();
+  const searchTypeId = generateId();
+
+  const [queryParameterStash, setQueryParameterStash] = useState<object>({});
+  const [searchFiltersHidden, setSearchFiltersHidden] = useState<boolean>(false);
+  const [searchWithinMsUnit, setSearchWithinMsUnit] = useState<string>(searchWithin.unit);
+  const [executeEveryMsUnit, setExecuteEveryMsUnit] = useState<string>(executeEvery.unit);
+  const [searchWithinMsDuration, setSearchWithinMsDuration] = useState<number>(searchWithin.duration);
+  const [executeEveryMsDuration, setExecuteEveryMsDuration] = useState<number>(executeEvery.duration);
+
+  const userCanViewLookupTables = useCallback(
+    () => isPermitted(currentUser.permissions, LOOKUP_PERMISSIONS),
+    [currentUser.permissions],
+  );
+
+  const validateQueryString = useCallback(
+    (queryString, streamIds, timeRange, timezone) => {
```

Comment:
We should add a type for these parameters. We also have some types that need to be included below.